File: python/trader/log_traders.py

from python.trader.trader import Trader
from python.trader.trader_parent import TraderParent


class LogTraders(TraderParent):
    """
    A trader that logs the update of the stock price.
    """

    def __init__(self, trader: Trader):
        """
        Initialize the LogTraders instance with a trader instance.
        """
        super(LogTraders, self).__init__(trader)

    def update(self, stock, price):
        """
        Update the stock price for the trader and log the update.
        """
        super(LogTraders, self).update(stock, price)
        self.log_update(stock, price)

    def log_update(self, stock, price):
        """
        Log the update of the stock price.
        """
        print(f"Trader {self.trader.__class__.__name__} updated {stock} to {price}")

File: java/src/main/java/org/nitk/hackathon/trader/LogTraders.java
package org.nitk.hackathon.trader;

public class LogTraders extends TraderParent {
    public LogTraders(Trader trader) {
        super(trader);
    }

    @Override
    public void update(String stock, double price) {
        super.update(stock, price);
        logUpdate(stock, price);
    }

    private void logUpdate(String stock, double price) {
        System.out.println("Trader " + trader.getClass().getSimpleName() + " updated " + stock + " to " + price);
    }
}